x-pack/plugins/index_management/__jest__/client_integration/index_template_wizard/template_edit.test.tsx

```diff
@@ -278,4 +278,66 @@ describe('<TemplateEdit />', () => {
       });
     });
   });
+
+  describe('legacy index templates', () => {
+    const legacyTemplateToEdit = fixtures.getTemplate({
+      name: 'legacy_index_template',
+      indexPatterns: ['indexPattern1'],
+      isLegacy: true,
+      template: {
+        mappings: {
+          my_mapping_type: {},
+        },
+      },
+    });
+
+    beforeAll(() => {
+      httpRequestsMockHelpers.setLoadTemplateResponse(legacyTemplateToEdit);
+    });
+
+    beforeEach(async () => {
+      await act(async () => {
+        testBed = await setup();
+      });
+
+      testBed.component.update();
+    });
+
+    it('persists mappings type', async () => {
+      const { actions } = testBed;
+      // Logistics
+      await actions.completeStepOne();
+      // Note: "step 2" (component templates) doesn't exist for legacy templates
```

Comment:
Side note - I think we should rename the helper actions at some point to be descriptive of the step, not the order in which they appear in the UI.